vm/src/builtins/pystr.rs

```diff
@@ -1197,6 +1242,38 @@ impl FindArgs {
     }
 }
 
+// Common reducer for forward and reverse str iterators.
```

Comment:
Since these two function aren't common, move the functionality in the methods now, no need to have it here anymore.